### ISSUE TITLE
Osmand suggest the wrong highway number

### ISSUE THREAD

**Issue by TheSnoozer** [osmandapp/OsmAnd#20679]
### Description

Hello osmand, maybe this is a special case, but I have noticed that Osmand suggest the wrong highway number, but with correct routing information.

### Steps to reproduce

https://osmand.net/map/navigate/?start=52.294426,12.935159&end=52.269176,12.907290&profile=car#14/52.2908/12.9243

### Actual result

In the top navigation bar Osmand suggest to use the "A10"
![photo_2024-08-25_09-02-08](https://github.com/user-attachments/assets/d1ab4a7b-a67d-4f18-810f-9d2544bd7e29)
the same applies in the details of the route "streight ahead on A10"
![photo_2024-08-25_09-01-31](https://github.com/user-attachments/assets/60fec58b-438d-4333-8b76-1e0424be2e1e)
Even the route navigation (under steps to reproduce) suggest to go streigt ahead on the "A10"
![image](https://github.com/user-attachments/assets/15815b1d-8ed2-454a-a268-d2aad03e7acc)

### Expected result

The navigation should suggest to go steight ahead on the **A9** and **not** the A10, as the A10 is actually the highway that branches off:
![image](https://github.com/user-attachments/assets/53828f0d-9296-4dac-9d9b-5d945a9ee02f)

Thank you for your time and consideration :-)

### Your Environment (required)

```
Device : RE513CL1
Brand : realme
Manufacturer : realme
Model : RMX3241
Product : RMX3241EEA
Build : RMX3241_11_F.11
Version : 13
App Version : OsmAnd+
Apk Version : 4.8.6 4806
```

**Comment by vshcherb** [comment]
Yes lane ifnormation is correct but it states that you're still driving on A10, so it doesn't suggest to keep driving on A10 later. As later we're going to split current street name and turn street name that confusion will be gone.

**Comment by TheSnoozer** [comment]
Ahhh I see thanks for the clarifications! Should I close this then, or wait until fixed? I don't mind you have plenty of bugs so if this is on yourplan anyways it would be fixed eventually :-)

Thanks for your support!
